falcon/api.py

```diff
@@ -103,6 +103,10 @@ def process_response(self, req, resp, resource, req_succeeded)
             to use in lieu of the default engine.
             See also: :ref:`Routing <routing>`.
 
+        router_search_method (str, optional): The name of a custom router method
```

Comment:
Up to you, really. 

I'm happy to write (c) and a ticket to track (b) if you think that's a good approach. 

Otherwise we can update https://github.com/falconry/falcon/issues/967 to track the 2.0 plan and leave this out for now.